### DISCUSSION TITLE
How do I authenticate without SDK?

### DISCUSSION THREAD

**Discussion by g1sbi** [pocketbase/pocketbase#2251] [Q&A]
I'm trying to authenticate in PB in a Swift app. I understand I have to send an `Authorization` header and the token, but I have no idea where my admin token is.

**Comment by danibjor** [comment]
When you post (email + password) to `http://your-server/api/auth-with-password` to authenticate, on success you get a response containing  containing your auth-token. This auth-token is used in the header on subsequent requests to fetch protected data. 

The payload/body on the initial request when authenticating should be in the form of 
`{ identity: "user@test.com", password: "super-secret" }`.

**Comment by ganigeorgiev** [comment]
@g1sbi Please check the admin web APIs doc - https://pocketbase.io/docs/api-admins/.

The admin token can be generated by sending `POST /api/admins/auth-with-password` request.

**Comment by g1sbi** [comment]
I'm probably encoding my data wrong. I wrote this 
```swift
 var request = URLRequest(url: URL(string: "http://127.0.0.1:8090/api/admins/auth-with-password")!)
        request.httpMethod = "POST"
        
        let authData = (email + ":" + password).data(using: .utf8)!.base64EncodedString()
        request.addValue("Basic \(authData)", forHTTPHeaderField: "Authorization")
```
But in the logs the request returns a `400` with:
```json
{
  "errorMessage": "identity: cannot be blank; password: cannot be blank."
}
```
Is the header formatted correctly? Should it not be encoded in base64?

**Reply by ganigeorgiev** [reply]
I'm sorry, I don't understand what you are doing. The name of the parameter is `identity`, not `email`. Also you don't need to send `Authorization: Basic` header for this request. Just the `identity` and `password` as body.

Please check again the "Auth with password" documentation in https://pocketbase.io/docs/api-admins/.

**Reply by TMBL-DEV** [reply]
There are literally docs in the pocketbase admin panel.
And a quick tip.

Make your requests in postman or insomnia so you can test the requests first and then code out the feature.

And in the worst case you can always read the requests from your network tab.

**Reply by g1sbi** [reply]
Yeah I'm sorry, I eventually found the Web API docs but they were kinda hidden away. Not an excuse but I did look before opening the discussion. Anyway, I managed to get the auth going, so I'll close this. Thanks for the tip I'll give that a try!

**Comment by ghost** [comment]
I am also very puzzled how to make my first simple REST API request. The only thing I see in the docs is:
<img width="731" alt="image" src="https://github.com/user-attachments/assets/9dce2beb-e2f1-4f2c-bb7b-f0c7b458f223">

but not how or where to find this `TOKEN`

**Reply by ghost** [reply]
UPDATE:

OK, found out that I had to authenticate via:
`http://127.0.0.1:8090/api/admins/auth-with-password`

body: `{ identity: "EMAIL", password: "PASSWORD" }`

and that responds with an admin token. It would be nice to know about it when creating first collection and trying to access it via API.

**Reply by ganigeorgiev** [reply]
@craftyme By default all new collections are "locked" (aka. admin only). You can adjust the access to your collection via the [API rules](https://pocketbase.io/docs/api-rules-and-filters/).

**Reply by ghost** [reply]
@ganigeorgiev thanks you, I wasn't aware! That's helpful too!

**Reply by abhranil26** [reply]
> UPDATE:
> 
> OK, found out that I had to authenticate via: `http://127.0.0.1:8090/api/admins/auth-with-password`
> 
> body: `{ identity: "EMAIL", password: "PASSWORD" }`
> 
> and that responds with an admin token. It would be nice to know about it when creating first collection and trying to access it via API.

Does this endpoint work anymore? I am getting 404 error in v0.26.3 Release

**Reply by abhranil26** [reply]
nvm, the new endpoint is /api/collections/_superusers/auth-with-password
